crates/oxc_language_server/src/main.rs

```diff
@@ -180,8 +188,38 @@ impl LanguageServer for Backend {
         }
     }
 
-    async fn did_change_watched_files(&self, _params: DidChangeWatchedFilesParams) {
+    async fn did_change_watched_files(&self, params: DidChangeWatchedFilesParams) {
         debug!("watched file did change");
+        if self.options.lock().await.use_nested_configs {
+            let mut config_files = FxHashMap::default();
+            let existing_nested_configs = (*self.nested_configs.read().await).clone();
+            for (key, value) in existing_nested_configs {
+                config_files.insert(key.clone(), value.clone());
```

Comment:
instead of inserting one-by-one, maybe we can try extending the hash map all at once? this should be more efficient because it allocates the needed memory upfront: https://doc.rust-lang.org/nightly/std/collections/struct.HashMap.html#impl-Extend%3C(K,+V)%3E-for-HashMap%3CK,+V,+S%3E